crates/ty_python_semantic/src/types.rs

```diff
@@ -1003,6 +1003,26 @@ impl<'db> Type<'db> {
                 self_typevar == other_typevar
```

Comment:
This arm is removable because we (now, as of this PR) handle it in equivalence, which is already checked above.

Reply:
In fact, it was already handled prior to this PR in equivalence! The last branch of the `match` statement handles it. The one that you and I both added in our separate PRs was redundant.

I added tests (as you suggested) that demonstrate this.